vllm/distributed/kv_events.py

```diff
@@ -353,12 +353,16 @@ def create(
         cls, config: KVEventsConfig | None, data_parallel_rank: int = 0
     ) -> EventPublisher:
         """Create publisher from a config mapping."""
-        if not config:
+        if (
+            not config
+            or not config.enable_kv_cache_events
+            or config.publisher == "null"
+        ):
             return NullEventPublisher()
 
         config_dict = asdict(config)
 
-        kind = config_dict.pop("publisher", "null")
+        kind = config_dict.pop("publisher")
```

Comment:
Just an observation, not a problem with this PR - if we ever add another publisher other than `ZmqEventPublisher` it will be required to accept all the members of `KEventsConfig` as constructor parameters, which doesn't make sense